### PR TITLE
Migrate to non-transitive R classes

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Switch to [non-transitive R classes](https://blog.blundellapps.co.uk/speed-up-your-build-non-transitive-r-files/), which became the default starting in Android Gradle plugin 8.0. This fixes build issues with the CI pipeline and helps speed up the build process.

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Isira-Seneviratne** [comment]
The SonarCloud warnings are due to the app using private ExoPlayer strings.

**Comment by Profpatsch** [comment]
Thank you, I rebased the change onto current `dev`.

**Comment by Stypox** [comment]
How did you choose the replacements? To me they all seem random. For example, why `com.google.android.material.R.attr.colorSecondary`, but instead `androidx.appcompat.R.attr.colorAccent`?

File: app/src/main/java/org/schabi/newpipe/error/ErrorUtil.kt [hunk 1]
```diff
@@ -54,7 +54,11 @@ class ErrorUtil {
          */
         @JvmStatic
         fun showSnackbar(context: Context, errorInfo: ErrorInfo) {
-            val rootView = if (context is Activity) context.findViewById<View>(R.id.content) else null
+            val rootView = if (context is Activity) {
+                context.findViewById<View>(androidx.constraintlayout.widget.R.id.content)
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/error/ErrorUtil.kt:58]: For example, this is wrong, and should instead be `android.R.id.content`.

**Comment by Stypox** [comment]
Replaced by https://github.com/TeamNewPipe/NewPipe/pull/10918 for the reasons stated above. @Isira-Seneviratne please review/approve https://github.com/TeamNewPipe/NewPipe/pull/10918, thanks :-).

**Pull request closed without merging**